cirq-google/cirq_google/ops/coupler_pulse.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""An instance of FSimGate that works naturally on Google's Sycamore chip"""
+
+from typing import Optional, Tuple, TYPE_CHECKING
+
+import numpy as np
+
+from cirq import ops, protocols, value
+
+if TYPE_CHECKING:
+    import cirq
+
+
+_MIN_DURATION = value.Duration(nanos=0)
+_MAX_DURATION = value.Duration(nanos=100)
+
+
+class CouplerPulse(ops.gate_features.TwoQubitGate):
+    """Tunable pulse for entangling adjacent qubits.
+
+    For experimental usage only.
+
+    This operation sends a trapezoidal pulse to the coupler between two
+    adjacent qubits placed in resonance.
+
+    Note that this gate does not have a unitary matrix and must be
+    characterized by the user in order to determine its effects.
+
+    Args:
+            hold_time: Length of the 'plateau' part of the coupler trajectory.
+            coupling_MHz: Target qubit-qubit coupling reached at the plateau.
+            rise_time: Full width of the smoothstep rise/fall.
+            padding_time: Symmetric padding around the coupler pulse.
+    """
+
+    def __init__(
+        self,
+        hold_time: value.Duration,
+        coupling_mhz: float,
```

Comment:
nit: internally we generally use standard case for units, even when it clashes with variable naming conventions, so this would be `coupling_MHz`. We haven't really established a convention one way or the other in cirq since we don't use many units like this in names, but my preference would be to stick with our internal convention.

Reply:
This violates case conventions per Google python style guide, so I don't think it is a good convention to follow.

Reply:
I would stick with the Google python style guide too here. If @maffoo you feel strongly about it, we can bring to the Cirq Cynque and get a vote on it.